nsxt/resource_nsxt_policy_ipsec_vpn_session_test.go

```diff
@@ -489,3 +632,84 @@ resource "nsxt_policy_ipsec_vpn_session" "test" {
 			attrMap["authentication_mode"], attrMap["compliance_suite"], attrMap["peer_address"], attrMap["peer_id"],
 			attrMap["psk"], attrMap["connection_initiation_mode"], attrMap["sources"], attrMap["destinations"], attrMap["action"])
 }
+
+func testAccNsxtPolicyIPSecVpnSessionPolicyBasedTemplateTier1(createFlow bool) string {
+	var attrMap map[string]string
+	if createFlow {
+		attrMap = accTestPolicyIPSecVpnSessionPolicyBasedCreateAttributes
+	} else {
+		attrMap = accTestPolicyIPSecVpnSessionPolicyBasedUpdateAttributes
+	}
+	return testAccNsxtPolicyEdgeClusterReadTemplate(getEdgeClusterName()) +
+		testAccNsxtPolicyTier1WithEdgeClusterForVPN("test") +
+		testAccNsxtPolicyIPSecVpnSessionPreConditionTemplate(false) +
+		fmt.Sprintf(`
+resource "nsxt_policy_ipsec_vpn_session" "test" {
+	display_name               = "%s"
+	description                = "%s"
+	ike_profile_path           = nsxt_policy_ipsec_vpn_ike_profile.test.path
+	tunnel_profile_path        = nsxt_policy_ipsec_vpn_tunnel_profile.test.path
+	dpd_profile_path		   = nsxt_policy_ipsec_vpn_dpd_profile.test.path
+	local_endpoint_path		   = nsxt_policy_ipsec_vpn_local_endpoint.test.path
+	enabled                    = "%s"
+	service_path               = nsxt_policy_ipsec_vpn_service.test.path
+	vpn_type                   = "%s"
+	authentication_mode        = "%s"
+	compliance_suite           = "%s"
+	peer_address               = "%s"
+	peer_id                    = "%s"
+	psk                        = "%s"
+	connection_initiation_mode = "%s"
+
+	rule {
+		sources             = ["%s"]
+		destinations        = ["%s"]
+		action              = "%s"
+	  }
+
+	tag {
+	scope = "scope1"
+	tag   = "tag1"
+	  }
+}`, attrMap["display_name"], attrMap["description"], attrMap["enabled"], attrMap["vpn_type"],
+			attrMap["authentication_mode"], attrMap["compliance_suite"], attrMap["peer_address"], attrMap["peer_id"],
+			attrMap["psk"], attrMap["connection_initiation_mode"], attrMap["sources"], attrMap["destinations"], attrMap["action"])
+}
+
+func testAccNsxtPolicyIPSecVpnSessionRouteBasedTemplateTier1(createFlow bool) string {
```

Comment:
This seems very similar to `testAccNsxtPolicyIPSecVpnSessionRouteBasedTemplate`, could we avoid the code duplication with extra parameter?

Reply:
done